rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -261,7 +264,8 @@ rmw_destroy_client(rmw_node_t * node, rmw_client_t * client)
     }
     delete info;
   }
-  delete (client);
+  rmw_free(const_cast<char *>(client->service_name));
```

Comment:
I guess using that logic, I should check for NULL in the destroy() as well.  I'll do that.